test/boost/schema_change_test.cc

```diff
@@ -364,6 +364,47 @@ SEASTAR_TEST_CASE(test_combined_column_add_and_drop) {
     });
 }
 
+SEASTAR_TEST_CASE(test_concurrent_table_creation_with_different_schema) {
```

Comment:
I think this is good as a unit test, but it seem to be to be really emulating "concurrent table creation". It seems (unless I'm missing something) to emulate the case where two table creation mutations were merged without serializing the operations - something which anyway shouldn't happen in the new Raft code. Maybe it's worth commenting (if it's true...) that the situation simulated in this test can only be reached in the old pre-raft implementation?

Reply:
Yes, it's for the pre-raft world.

Reply:
Maybe you can say this in the commit message and also in a comment in the test (saying it emulates what happens in a pre-raft concurrent update) and possibly - also in the actual code? A year from now, when we only have Raft, nobody will remember why this test tests a situation that cannot happen, or why that code even exists because it fixes a situation that cannot happen with Raft.

Come to think of it, if in 2 our 5 years we do schema changes only with Raft (and remove the non-raft option), should the new code you add in this patch (which adds tombstones, etc.) be deleted because it's not needed?

Reply:
It probably could be deleted, but in 5 years nobody will dare.

Reply:
That's exactly why I think we should have comments in the code and test saying why this specific code is needed (and that it isn't needed if schema modifications are serialized by another means, such as raft). I think you did have a comment, but it wasn't explicit about the new code not being needed after raft.